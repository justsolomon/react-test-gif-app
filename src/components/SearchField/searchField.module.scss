@import "../../assets/scss/base/variables";

.search-input {
  &__container {
    display: flex;
    align-items: center;

    button {
      height: 50px;
      width: 60px;
      display: flex;
      align-items: center;
      justify-content: center;
      border-radius: $border-radius--lg;
      border-top-left-radius: 0;
      border-bottom-left-radius: 0;
    }
    svg {
      height: 1em;
      width: 1em;
    }
  }
  &__field {
    height: 50px;
    width: 100%;
    padding: 0px 1rem;
    font-size: $font-size--md;
    border: 1px solid $bg-light--secondary;
    border-radius: $border-radius--lg;
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
    box-shadow: 0 5px 22px rgba(0, 0, 0, 0.09);
    outline: none;

    &:hover {
      box-shadow: 0 5px 22px rgba(0, 0, 0, 0.13);
    }

    &:focus {
      box-shadow: 0 5px 22px rgba(0, 0, 0, 0.15);
    }
  }

  @media #{$media-mobile-sm} {
    &__container button,
    &__field {
      height: 45px;
    }

    &__container button {
      width: 55px;
      border-top-right-radius: $border-radius--md;
      border-bottom-right-radius: $border-radius--md;
    }

    &__field {
      font-size: $font-size--base;
      border-top-left-radius: $border-radius--md;
      border-bottom-left-radius: $border-radius--md;
    }
  }
}
