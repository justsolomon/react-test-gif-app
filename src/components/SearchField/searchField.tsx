import { Button } from "components/global/Button";
import styles from "./searchField.module.scss";
import { ReactComponent as SearchIcon } from "assets/vectors/search.svg";

interface ISearchField {
  query: string;
  updateQuery: React.ChangeEventHandler<HTMLInputElement>;
}

const SearchField = ({ query, updateQuery }: ISearchField): JSX.Element => {
  return (
    <div className={styles["search-input__container"]}>
      <input
        className={styles["search-input__field"]}
        placeholder="Search for GIF"
        value={query}
        onChange={updateQuery}
      />
      <Button type="submit">
        <SearchIcon />
      </Button>
    </div>
  );
};

export default SearchField;
