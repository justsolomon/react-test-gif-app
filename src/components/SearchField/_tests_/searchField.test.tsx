import { render } from "@testing-library/react";
import SearchField from "../searchField";

describe("<SearchField />", () => {
  it("should render the SearchField component properly", () => {
    const { asFragment } = render(
      <SearchField
        query="test"
        updateQuery={(e) => console.log(e.target.value)}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
