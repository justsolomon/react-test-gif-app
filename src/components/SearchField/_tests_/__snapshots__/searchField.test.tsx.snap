// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> should render the SearchField component properly 1`] = `
<DocumentFragment>
  <div
    class="search-input__container"
  >
    <input
      class="search-input__field"
      placeholder="Search for gif"
      value="test"
    />
    <button
      class="button  button--solid button--none "
      type="submit"
    >
      <svg>
        search.svg
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
